lib/routing/sfRoute.class.php

```diff
@@ -279,6 +270,45 @@ public function generate($params, $context = [], $absolute = false)
         return $url;
     }
 
+    /**
+     * Generates a consistent cache key for a route
+     */
+    protected function generateCacheKey(array $elements = [], bool $absolute = false): string
+    {
+        $elements = [
+            SharedCacheHelper::ROUTING_NAMESPACE,
+            $this->flattenArrayElements($elements),
+        ];
+
+        if ($absolute) {
+            $elements[] = 'Absolute';
+        }
+
+        return SharedCacheHelper::getNamespace($elements);
+    }
+
+    /**
+     * Recursive method to pack a key => value pair array down into a flat
+     * numerically-indexed array, with a consistent ordering of elements.
+     */
+    protected function flattenArrayElements(array $elements = []): array
+    {
+        $flattened = [];
+        ksort($elements);
+        foreach ($elements as $key => $value) {
+            $flattened[] = $key;
+
+            if (is_array($value)) {
+                $flattened[] = $this->flattenArrayElements($value);
```

Comment:
This feels unlikely, but I wanted to add some degree of protection in just in case.